src/components/learner-credit-management/data/utils.js

```diff
@@ -551,3 +556,111 @@ export const isLmsBudget = (
   activeIntegrationsLength,
   isUniversalGroup,
 ) => activeIntegrationsLength > 0 && isUniversalGroup;
+
+/**
+ * Determines if the course has already started. Mostly used around text formatting for tense
+ *
+ * @param date
+ * @returns {boolean}
+ */
+export const isDateBeforeToday = date => dayjs(date).isBefore(dayjs());
+
+/**
+ * Filters assignable course runs based on the following criteria:
+ *  - If hasEnrollBy, we return the soonest of two dates: The subsidy expiration date - refund threshold OR today
+ *    offset by the 90-day allocation threshold for an assignment denoted as isEligibleForEnrollment
+ *  - If isLateRedemptionEnabled, we consider only the isLateEnrollmentEligible field returned by Algolia for
+ *    each run.
+ *
+ *  Based on the above criteria, if isLateRedemptionAllowed is false, filter on if the course run isActive AND
+ *  isEligibleForEnrollment
+ *
+ *  Furthermore, we return assignable course runs sorted by the enrollBy date (soonest to latest)
+ *
+ * @param courseRuns
+ * @param subsidyExpirationDatetime
+ * @param isLateRedemptionAllowed
+ * @returns {*}
+ */
+export const getAssignableCourseRuns = ({ courseRuns, subsidyExpirationDatetime, isLateRedemptionAllowed }) => {
+  const clonedCourseRuns = courseRuns.map(courseRun => ({
+    ...courseRun,
+    enrollBy: courseRun.hasEnrollBy ? dayjs.unix(courseRun.enrollBy).toISOString() : null,
+    upgradeDeadline: dayjs.unix(courseRun.upgradeDeadline).toISOString(),
+  }));
+  const assignableCourseRunsFilter = ({
+    enrollBy, isActive, hasEnrollBy = false, isLateEnrollmentEligible = false,
+  }) => {
+    let isEligibleForEnrollment = true;
+    if (hasEnrollBy) {
+      isEligibleForEnrollment = dayjs(enrollBy).isBefore(
+        Math.min(
+          dayjs(subsidyExpirationDatetime).subtract(MAX_ALLOWABLE_REFUND_THRESHOLD_DAYS, 'days').toDate(),
+          dayjs().add(DAYS_UNTIL_ASSIGNMENT_ALLOCATION_EXPIRATION, 'days').toDate(),
+        ),
+      );
```

Comment:
You sure just changing this to min is sufficient? I ask again, what's the goal? The docstring was updated to match this logic, but still the desired outcome is not explicitly stated.  If the goal is as I described it previously, then I think this is still incorrect because it guarantees that learners ALWAYS have less than 90 days to accept an assignment.